infra/enrolment-records/room-store/src/main/java/com/simprints/infra/enrolment/records/room/store/SubjectDao.kt

```diff
@@ -0,0 +1,46 @@
+package com.simprints.infra.enrolment.records.room.store
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.MapColumn
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import androidx.room.RawQuery
+import androidx.sqlite.db.SupportSQLiteQuery
+import com.simprints.infra.enrolment.records.room.store.models.DbBiometricTemplate
+import com.simprints.infra.enrolment.records.room.store.models.DbSubject
+import com.simprints.infra.enrolment.records.room.store.models.DbSubjectTemplateFormatMap
+import com.simprints.infra.enrolment.records.room.store.models.SubjectBiometrics
+
+@Dao
+interface SubjectDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertSubject(subject: DbSubject)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertBiometricSamples(samples: List<DbBiometricTemplate>)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertSubjectTemplateMapping(mapping: DbSubjectTemplateFormatMap)
+
+    @Query("DELETE FROM DbSubject WHERE subjectId = :subjectId")
+    suspend fun deleteSubject(subjectId: String)
+
+    @Query("DELETE FROM DbBiometricTemplate WHERE uuid = :uuid")
+    suspend fun deleteBiometricSample(uuid: String)
+
+    @RawQuery
+    suspend fun deleteSubjects(query: SupportSQLiteQuery): Int
+
+    @Query("SELECT * FROM DbSubject WHERE subjectId = :subjectId")
+    suspend fun getSubject(subjectId: String): SubjectBiometrics?
+
+    @RawQuery
+    suspend fun loadSubjects(query: SupportSQLiteQuery): List<SubjectBiometrics>
```

Comment:
It is a bit concerning that query building is disjointed from the returning model definition; sooner or later, a bug will be caused by changing one place and forgetting another.

I am not sure what would be the best way to fix it, tho.